src/main/java/net/neoforged/neoforge/resource/ContextAwareReloadListener.java

```diff
@@ -5,15 +5,65 @@
 
 package net.neoforged.neoforge.resource;
 
+import com.google.gson.JsonElement;
+import com.mojang.serialization.JsonOps;
 import net.minecraft.core.RegistryAccess;
+import net.minecraft.resources.RegistryOps;
+import net.minecraft.server.packs.resources.PreparableReloadListener;
+import net.minecraft.server.packs.resources.SimplePreparableReloadListener;
+import net.neoforged.neoforge.common.conditions.ConditionalOps;
 import net.neoforged.neoforge.common.conditions.ICondition;
+import net.neoforged.neoforge.common.conditions.ICondition.IContext;
+import org.jetbrains.annotations.ApiStatus;
 
-public abstract class ContextAwareReloadListener {
+/**
+ * Reload listeners that descend from this class will have the reload context automatically populated when it is available.
+ * <p>
+ * The context is guaranteed to be available for the duration of {@link PreparableReloadListener#reload}.
+ * <p>
+ * For children of {@link SimplePreparableReloadListeners}, it will be available during both {@link SimplePreparableReloadListener#prepare prepare()} and {@link SimplePreparableReloadListener#apply apply()}.
+ */
+public abstract class ContextAwareReloadListener implements PreparableReloadListener {
+    /**
+     * TODO: Make private
+     * 
+     * @deprecated Use {@link #getContext()}
+     */
+    @Deprecated
     protected ICondition.IContext conditionContext = ICondition.IContext.EMPTY;
+
+    /**
+     * TODO: Make private
+     * 
+     * @deprecated Use {@link #getRegistryAccess()}
+     */
+    @Deprecated
     protected RegistryAccess registryAccess = RegistryAccess.EMPTY;
```

Comment:
Why though?

Reply:
The API surface we want to expose here is just a view to the context/reg access - we don't need to expose this field as it breaks encapsulation of our internal data (since the setter is internal, the fields should also be internal).